client/src/node.rs

```diff
@@ -123,6 +136,49 @@ impl Node {
         Ok(())
     }
 
+    async fn get_payloads(&self, start_slot: u64, end_slot: u64) -> Result<Vec<ExecutionPayload>, NodeError> {
+        let payloads_fut = (start_slot..end_slot).rev().into_iter().map(|slot| async move {
+            let self_ref = self;
+            let inner_closure = async move {
+                let payload = self_ref
+                    .consensus
+                    .get_block_from_rpc(slot)
+                    .await
+                    .map_err(NodeError::ConsensusPayloadError)?
+                    .body
+                    .execution_payload;
+                Ok::<ExecutionPayload, NodeError>(payload)
+            };
+            inner_closure.await
+        });
+        let mut prev_parent_hash: Option<Bytes32> = None;
+        let mut payloads: Vec<ExecutionPayload> = Vec::new();
+        for result in join_all(payloads_fut).await {
+            if result.is_err() {
+                // If fetching any payload failed we only backfill up to that payload
```

Comment:
Note that this can result in "holes" in the backfilled blocks, because at the next attempt backfilling will only start at the latest block from the previous update.

An alternative here would be to check all slots since `history_size` blocks ago in the `self.payloads` mapping and attempting to backfill any gaps. I can refactor it accordingly if it's worth it.